widgetastic_manageiq/__init__.py

```diff
@@ -5368,3 +5368,59 @@ def select(self, name):
             self.browser.element(self.BUTTON.format(quote(name))).click()
             return True
         return False
+
+
+class RolesSelector(Widget):
+    """This widget for custom button role selection"""
+
+    ROOT = "//div[@class='col-md-10']//table"
+    CELLS = "./tbody/tr/td"
+    INPUTS = "./tbody/tr/td/input"
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+
+    @property
+    def _elements(self):
+        br = self.browser
+        return {
+            br.text(name): el for name, el in zip(br.elements(self.CELLS), br.elements(self.INPUTS))
+        }
+
+    @property
+    def roles(self):
+        """all available roles"""
+        return list(self._elements.keys())
+
+    @property
+    def currently_selected(self):
+        """Currently selected roles"""
+        return [r for r in self.roles if self.role_selected(r)]
+
+    def role_selected(self, role):
+        """Check role is selected or not"""
+        if isinstance(role, six.string_types):
+            role = self._elements.get(role)
+        return self.browser.is_selected(role)
+
+    def select_all(self):
+        """Select all available roles"""
+        for el in self._elements.values():
+            if not self.role_selected(el):
+                el.click()
+
+    def clear_all(self):
+        """Clear all selected roles"""
+        for el in self._elements.values():
+            if self.role_selected(el):
+                el.click()
+
+    def fill(self, roles):
```

Comment:
Can you explain why is it `fill` method. As I see, it selects specific roles unlike `select_all`

Reply:
@lina-nikiforova as per widgetatic guidelines widget have `fill` and `read` and I'm not selecting all roles... I want to select specific roles.
1. `select_all` will select all available roles
2. `fill` method will select specific roles and it will help if we use `view.fill`

Reply:
@digitronik yep, I understood what it does but had a questioned the name. Thanks for the explanation, I think all good :)